libminifi/src/controllers/keyvalue/AbstractCoreComponentStateManagerProvider.cpp

```diff
@@ -24,6 +24,10 @@
 #include "rapidjson/stringbuffer.h"
 #include "rapidjson/writer.h"
 
+#ifdef GetObject
+#undef GetObject
+#endif /* GetObject -> GetObject[AW] on windows */
```

Comment:
I don't think we need the `#ifdef`; also I would include a comment everywhere, not just here.

```suggestion
// windows.h #defines GetObject = GetObjectA or GetObjectW, which conflicts with rapidjson
#undef GetObject
```